custom_components/tuya_local/devices/kcvents_vt501_fan.yaml

```diff
@@ -0,0 +1,153 @@
+  translation_only_key: fan_with_presets
+  dps:
+    - id: 1
+      type: boolean
+      name: switch
+    - id: 101
+      type: boolean
+      name: speed
+      default: false
+      mapping:
+        - dps_val: true
+          value: 33
+        - dps_val: dummy_speed1
+          value: 66
+          value_redirect: speed2
+        - dps_val: dummy_speed2
+          value: 100
+          value_redirect: speed3
+        - dps_val: false
+          value_redirect: speed2
+    - id: 102
+      type: boolean
+      name: speed2
+      default: false
+      mapping:
+        - dps_val: true
+          value: 66
+        - dps_val: false
+          value_redirect: speed3
+      hidden: true
+    - id: 103
+      type: boolean
+      name: speed3
+      default: false
+      mapping:
+        - dps_val: true
+          value: 100
+        - dps_val: false
+          value: 0
+      hidden: true
+    - id: 105
+      type: boolean
+      name: preset_mode
+      default: false
+      mapping:
+        - dps_val: true
+          value: "Low Humidity"
+        - dps_val: dummy_mode1
+          value: "Medium Humidity"
+          value_redirect: preset_mode1
+        - dps_val: dummy_mode2
+          value: "High Humidity"
+          value_redirect: preset_mode2
+        - dps_val: dummy_mode3
+          value: "Fresh Air Mode"
+          value_redirect: preset_mode3
+        - dps_val: dummy_mode4
+          value: "Exhausted Air Mode"
+          value_redirect: preset_mode4
+        - dps_val: dummy_mode5
+          value: "Regenerate Mode"
+          value_redirect: preset_mode5
+        - dps_val: false
+          value_redirect: preset_mode1
+    - id: 106
+      type: boolean
+      name: preset_mode1
+      default: false
+      mapping:
+        - dps_val: true
+          value: "Medium Humidity"
+        - dps_val: false
+          value_redirect: preset_mode2
+      hidden: true
+    - id: 107
+      type: boolean
+      name: preset_mode2
+      default: false
+      mapping:
+        - dps_val: true
+          value: "High Humidity"
+        - dps_val: false
+          value_redirect: preset_mode3
+      hidden: true
+    - id: 111
+      type: boolean
+      name: preset_mode3
+      default: false
+      mapping:
+        - dps_val: true
+          value: "Fresh Air Mode"
+        - dps_val: false
+          value_redirect: preset_mode4
+      hidden: true
+    - id: 112
+      type: boolean
+      name: preset_mode4
+      default: false
+      mapping:
+        - dps_val: true
+          value: "Exhausted Air Mode"
+        - dps_val: false
+          value_redirect: preset_mode5
+      hidden: true
+    - id: 113
+      type: boolean
+      name: preset_mode5
+      default: false
+      mapping:
+        - dps_val: true
+          value: "Regenerate Mode"
+        - dps_val: false
+          value: NONE
+      hidden: true
+secondary_entities:
+  - entity: sensor
+    class: humidity
+    name: current_humidity
+    readonly: true
```

Comment:
readonly is not valid here, and is not needed for sensors as they are always readonly.